src/Components/Form.js

```diff
@@ -1,19 +1,39 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addbook } from '../redux/books/booksSlice';
 
 function Form() {
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [category, setCategory] = useState('');
+
+  const dispatch = useDispatch();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch(addbook({ title, author, category }));
+    setTitle('');
+    setAuthor('');
+    setCategory('');
+  };
+
   return (
 
-    <form>
+    <form onSubmit={handleSubmit}>
       <label htmlFor="book">
-        <input id="book" name="book" type="text" placeholder="Book title" />
+        <input id="book" name="book" type="text" placeholder="Book title" onChange={(e) => setTitle(e.target.value)} />
+      </label>
+
+      <label htmlFor="author">
+        <input id="author" name="author" type="text" placeholder="Author" onChange={(e) => setAuthor(e.target.value)} />
```

Comment:
**OPTIONAL** - After submitting your form I suggest resetting your form input, though I see you have set your state for the input to an empty string, you aren't setting the `value` attribute to the current value of your state. Please fix this 🙏